konveyor/pelorus
# feat: Add pre-commit framework

## Describe the behavior changes introduced in this PR

Adds [pre-commit framework](https://pre-commit.com/) to Pelorus.

## Linked Issues

resolves #681
resolves #692

related to #780

## Testing Instructions

Run `make dev-env` (this will run `pre-commit install` and setup pre-commit to you). Then, run `pre-commit run --all-files` to run pre-commit hooks against all files of Pelorus. Then, change a file and add it and commit it to see it in action.

If one of the hooks fails (defined in `.pre-commit-config.yaml` file), the commit is aborted. This can be bypassed by running `git commit` command with the `-n/--no-verify` flag.
